### PR TITLE
Use GitHub API instead of screen scraping.

### PR DESCRIPTION
**Pull request by MarkSRobinson**
TinkerTime does screen scraping which is fragile.  This changes to use the GitHub API proper.

### CODE REVIEW

**Comment by apemanzilla** [comment]
You'll have to edit the unit tests to get Travis to stop complaining.

**Comment by MarkSRobinson** [comment]
There we go.  Cleaned up the build.

**Comment by oharaandrew314** [comment]
I thought about using this, but the unauthenticated github api has a limit
on the number of requests.  I think its better to just scrape.

**Comment by MarkSRobinson** [comment]
It does have limit, but it is decently high (60/hr) and we're caching responses already.  If people need more, we can authenticate against GitHub.

**Comment by oharaandrew314** [comment]
Oh right, I forgot about the caching.  I'll look at your PR and maybe keep
a backup of the original scraper.

**Comment by apemanzilla** [comment]
If caching is a problem just add a changing get parameter such as "&time=" + System.getCurrentMillis()

File: src/aohara/tinkertime/crawlers/GithubCrawler.java [hunk 1]
```diff
@@ -53,33 +66,43 @@ public Document getPage(URL url) throws IOException {
 		return super.getPage(url);
 	}
 	
-	private Element getLatestReleaseElement() throws IOException {
-		Document doc = getPage(getApiUrl());
-		return doc.select("div.label-latest").first();
+	private JsonObject getLatestReleaseElement() throws IOException {
+        JsonElement doc = getPage(getApiUrl());
+        for(int index = 0; index < doc.getAsJsonArray().size(); index++) {
```
Comment (oharaandrew314) [inline src/aohara/tinkertime/crawlers/GithubCrawler.java]: Can you not use a for-each loop here?
```
for (JsonObject jsonObject : doc.getAsJsonArray()){
    // stuff
}
```

File: testRes/json/ProceduralFairings.json [hunk 2]
```diff
@@ -0,0 +1,842 @@
+[
```
Comment (oharaandrew314) [inline testRes/json/ProceduralFairings.json]: Please [minify](http://www.willpeavy.com/minifier/) this

File: src/aohara/tinkertime/crawlers/GithubCrawler.java [hunk 3]
```diff
@@ -53,33 +66,43 @@ public Document getPage(URL url) throws IOException {
...
-		for (Element assetEle : assetElements){
-			assets.add(new Asset(
-					assetEle.attr("href").substring(assetEle.attr("href").lastIndexOf('/') + 1),
-					new URL(assetEle.absUrl("href"))
-			));
-		}
+
+        JsonArray jsonArray = assetsElement.get("assets").getAsJsonArray();
+
+        for(int index = 0; index < jsonArray.size(); index++) {
```
Comment (oharaandrew314) [inline src/aohara/tinkertime/crawlers/GithubCrawler.java]: for-each here
```
for (JsonElement jsonEle : jsonArray) {
    // stuff
}
```

File: src/aohara/tinkertime/crawlers/CrawlerFactory.java [hunk 4]
```diff
@@ -4,6 +4,7 @@
 import java.net.URL;
 import java.util.Arrays;
 
+import com.google.gson.JsonElement;
 import org.jsoup.nodes.Document;
 
 import com.google.gson.JsonObject;
```
Comment (oharaandrew314) [inline src/aohara/tinkertime/crawlers/CrawlerFactory.java]: Unused import

File: src/aohara/tinkertime/crawlers/GithubCrawler.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package aohara.tinkertime.crawlers;
```
Comment (oharaandrew314) [inline src/aohara/tinkertime/crawlers/GithubCrawler.java]: Unused imports here are LinkedList, Document, and Element.

File: src/aohara/tinkertime/crawlers/pageLoaders/JsonLoader.java [hunk 6]
```diff
@@ -3,12 +3,16 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.Reader;
+import java.net.Authenticator;
+import java.net.PasswordAuthentication;
```
Comment (oharaandrew314) [inline src/aohara/tinkertime/crawlers/pageLoaders/JsonLoader.java]: Unused imports are Authenticator, PasswordAuthentication, UrlConnection, and JsonElement

File: test/test/util/MockCrawlerFactory.java [hunk 7]
```diff
@@ -4,6 +4,7 @@
 import java.io.InputStream;
 import java.net.URL;
 
+import com.google.gson.JsonElement;
```
Comment (oharaandrew314) [inline test/test/util/MockCrawlerFactory.java]: JsonObject is an unused import

**Comment by oharaandrew314** [comment]
Looks pretty good to me.  Please just fix the style issues I commented on, and I'll merge this.

File: testRes/json/ProceduralFairings.json (hunk 2 above)
Comment (MarkSRobinson) [inline testRes/json/ProceduralFairings.json]: I don't think we should for two reasons.

1) It's used for debugging and it really does need to be human readable.
2) It's a literal copy of what github produces.

File: src/aohara/tinkertime/crawlers/GithubCrawler.java (hunk 1 above)
Comment (MarkSRobinson) [inline src/aohara/tinkertime/crawlers/GithubCrawler.java]: Huh, turns out I can use it on that type.  Good to know.

File: testRes/json/ProceduralFairings.json (hunk 2 above)
Comment (oharaandrew314) [inline testRes/json/ProceduralFairings.json]: I just wish there was a way it wouldn't pollute the line-count on Github.

**Comment by oharaandrew314** [comment]
Testing again.  Some people have been complaining about some Github issues, so I may make a maintenance release with this.

**Comment by oharaandrew314** [comment]
Thanks @MarkSRobinson

**Pull request merged**